app/services/platform-apps/api/modules/scene-transitions.ts

```diff
@@ -0,0 +1,189 @@
+          propertiesManagerSettings: {
+            appId,
+            locked: shouldLock,
+          },
+          settings: {
+            ...settings,
+            path: this.platformAppsService.getAssetUrl(appId, path),
+          } as Dictionary<TObsValue>,
+        }),
+        toObsTransitionOptions(stingerTransitionDefaultOptions),
+      )(settings);
+
+      return this.transitionsService.createTransition(
+        ETransitionType.Stinger,
+        name,
+        transitionOptions,
+      );
+    }
+
+    throw new Error('Not Implemented');
+  }
+}
+
+const transitionTypeToObs = (type: string): ETransitionType => {
+  if (type === 'stinger') {
+    return ETransitionType.Stinger;
+  }
+};
+
+const audioFadeStyleToObs = (x: AudioFadeStyle): ObsAudioFadeStyle =>
+  x === 'fadeOut' ? ObsAudioFadeStyle.FadeOut : ObsAudioFadeStyle.CrossFade;
+
+const transitionPointTypeToObs = (transitionPoint: TransitionPointType): ETransitionPointType =>
+  transitionPoint === 'time' ? ETransitionPointType.Time : ETransitionPointType.Frame;
+
+const transforms = {
+  type: transitionTypeToObs,
+  audioFadeStyle: audioFadeStyleToObs,
+  shouldMonitorAudio: (shouldMonitor: boolean) => (shouldMonitor ? 1 : 0),
+  transitionPointType: transitionPointTypeToObs,
+};
+
+const stingerTransitionDefaultOptions: Partial<StingerTransitionOptions> = {
+  type: 'stinger',
+  transitionPointType: 'time',
+  shouldMonitorAudio: false,
+  audioFadeStyle: 'fadeOut',
+  shouldLock: false,
+};
+
+const obsKeyMapping = {
+  audioFadeStyle: 'audio_fade_style',
+  transitionPointType: 'tp_type',
+  shouldMonitorAudio: 'audio_monitoring',
+  transitionPoint: 'transition_point',
+  url: 'path',
+};
+
+const toObsTransitionKeys = (options: TransitionOptions): ITransitionCreateOptions => {
+  const result = {};
+
+  Object.keys(options).forEach(key => {
+    if (obsKeyMapping[key]) {
+      result[obsKeyMapping[key]] = options[key];
+    }
+  });
+
+  return result;
+};
+
+const assignDefaultOptions = (defaultOptions: Partial<TransitionOptions>) => (
+  options: Partial<TransitionOptions>,
+): TransitionOptions => ({ ...defaultOptions, ...options } as TransitionOptions);
+
+const toObsTransitionOptions: (
+  defaultOptions: Partial<TransitionOptions>,
+) => (options: Partial<TransitionOptions>) => ITransitionCreateOptions = defaultOptions =>
+  compose(
+    toObsTransitionKeys,
+    evolve(transforms),
+    assignDefaultOptions(defaultOptions),
+  );
+
+// TODO: type accordingly. I've looked at @types/ramda for this, but they're atrocious
+const evolve = (transforms: { [x: string]: (a: any) => any }) => (obj: object) => {
+  const result = {};
+
+  Object.keys(obj).forEach(key => {
+    const val = obj[key];
+    result[key] = transforms[key] ? transforms[key](val) : val;
+  });
+
+  return result;
+};
+
+const isVideo: (x: string) => boolean = compose(
+  (type: string | null) => /^video\/.*$/.test(type),
+  getType,
+  path.basename,
+);
```

Comment:
This whole module is written in a highly functional style that I'm finding very difficult to parse.  I understand this is the way you like to write code, but it's not the way the rest of this codebase is written.  For now I think we should refactor this to use a more OO style that matches the rest of the codebase.